scapy/layers/dot11.py

```diff
@@ -906,7 +906,7 @@ class Dot11EltCountry(Dot11Elt):
         ),
         ConditionalField(
             ByteField("pad", 0),
-            lambda pkt: (pkt.len + 1) % 2
+            lambda pkt: (len(pkt.descriptors) * 3) + 3 != pkt.len
```

Comment:
This should work and be enough:
```suggestion
            # The packet length is even (odd) iff it has an even (odd)
            # number of descriptors.
            lambda pkt: len(pkt.descriptors) % 2
```

Reply:
thanks for reviewing.

your suggestion is much more elegant and smarter, looks good for me as well.
I personally prefer mine because I am personally using this to read the packets it could be more robust in case some vendor may not keep this rule correctly, but too much concern? or correct me if misunderstood.

Again it looks good to me, thanks again for the review and suggestion.

Reply:
Sorry but let me correct my previous comment, your suggestion should be like
```python
lambda pkt: (len(pkt.descriptors) + 1) % 2
```
to make it work because there always is a `country_string`, which is 3 byte length.

Reply:
I'd rather keep the code simpler, and use the correct version you have submitted in your second comment.

I don't think the code you propose in this PR is "more robust", since it also relies on the size of each element being 3. Anyway, I don't think it is too much of a big deal.